src/cliffordsynthesis/CliffordSynthesizer.cpp

```diff
@@ -237,6 +251,29 @@
   }
 }
 
+void CliffordSynthesizer::twoQubitDepthOptimalSynthesis(
+    CliffordSynthesizer::EncoderConfig config, const std::size_t lower,
+    const std::size_t upper) {
+  // tQDepth-optimal synthesis is achieved by determining a timestep limit T
+  // such that there exists a solution with tQDepth T, but no solution with
+  // tQDepth T-1. This procedure uses an encoding (SQG - TQG) where in single
+  // and two qubit gates are allowed just one gate per timestep. This procedure
```

Comment:
This is a tiny bit misleading. The number of gates is not restricted per timestep